controllers/aggregated_clusteroperator_controller.go

```diff
@@ -62,40 +64,51 @@ func (r *AggregatedClusterOperatorReconciler) Reconcile(ctx context.Context, req
 	}
 	defer func() {
 		if err := coWriter.UpdateStatus(ctx, aggregatedCO, coBuilder.GetStatus()); err != nil {
-			log.Error(err, "error updating CO status")
+			log.Error(err, "error updating cluster operator status")
 		}
 	}()
 
 	// Set the default CO status conditions: Progressing=True, Degraded=False, Available=False
-	coBuilder.WithProgressing(configv1.ConditionTrue, "")
-	coBuilder.WithDegraded(configv1.ConditionFalse)
-	coBuilder.WithAvailable(configv1.ConditionFalse, "", "")
+	// TODO: always set a reason (message is optional, but desirable)
+	coBuilder.WithProgressing(metav1.ConditionTrue, "", "")
+	coBuilder.WithDegraded(metav1.ConditionFalse, "", "")
+	coBuilder.WithAvailable(metav1.ConditionFalse, "", "")
 	coBuilder.WithVersion("operator", r.ReleaseVersion)
-	coBuilder.WithRelatedObject("", "namespaces", "", r.SystemNamespace)
-	coBuilder.WithRelatedObject("platform.openshift.io", "platformoperators", "", "")
-	coBuilder.WithRelatedObject("core.rukpak.io", "bundles", "", "")
-	coBuilder.WithRelatedObject("core.rukpak.io", "bundledeployments", "", "")
+	coBuilder.WithRelatedObject(configv1.ObjectReference{Group: "", Resource: "namespaces", Name: r.SystemNamespace})
+
+	// NOTE: Group and resource can be referenced without name/namespace set, which is a signal
+	// that _ALL_ objects of that group/resource are related objects. This is useful for
+	// must-gather automation.
+	coBuilder.WithRelatedObject(configv1.ObjectReference{Group: platformv1alpha1.GroupName, Resource: "platformoperators"})
+
+	// TODO: move platform operator ownership of rukpak objects out prior to rukpak or PO GA.
+	coBuilder.WithRelatedObject(configv1.ObjectReference{Group: rukpakv1alpha1.GroupVersion.Group, Resource: "bundles"})
+	coBuilder.WithRelatedObject(configv1.ObjectReference{Group: rukpakv1alpha1.GroupVersion.Group, Resource: "bundledeployments"})
```

Comment:
We could also move this logic to a method/function that adds these related objects to the builder instance.